MotionMark/resources/runner/motionmark.js

```diff
@@ -116,15 +116,20 @@
             var regressionOptions = { desiredFrameLength: desiredFrameLength };
             if (profile)
                 regressionOptions.preferredProfile = profile;
+
+            var numSamples = maxIndex - minIndex + 1;
+            var regressionData = new Array(numSamples);
+            for (var i = 0; i < numSamples; ++i) {
+                var sample = series.data[i + minIndex];
+                regressionData[i] = [ sample[complexityIndex], sample[frameLengthIndex] ];
+            }
```

Comment:
I originally wrote it with a `slice()` but this both makes an extra array copy, and in a future change I'll be skipping some samples so that will turn it into two copies (slice followed by filter). This seems simpler.

Reply:
I'll use your suggestion.